hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/ArrayDataSerializableFactoryTest.java

```diff
@@ -0,0 +1,54 @@
+package com.hazelcast.internal.serialization.impl;
+
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.nio.serialization.SampleIdentifiedDataSerializable;
+import com.hazelcast.util.ConstructorFunction;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * ArrayDataSerializableFactory Tester.
+ *
+ */
+public class ArrayDataSerializableFactoryTest {
+
+    ArrayDataSerializableFactory factoryWithNonNullArray;
+    ArrayDataSerializableFactory factoryWithNonNullZeroArray;
+    ArrayDataSerializableFactory factoryWithNullArray;
+
+    @Before
+    public void before() throws Exception {
+        ConstructorFunction<Integer, IdentifiedDataSerializable>[] ctorArray = new ConstructorFunction[1];
+
+        ctorArray[0] = new ConstructorFunction<Integer, IdentifiedDataSerializable>() {
+            @Override
+            public IdentifiedDataSerializable createNew(Integer arg) {
+                return new SampleIdentifiedDataSerializable();
+            }
+        };
+
+        factoryWithNonNullArray = new ArrayDataSerializableFactory(ctorArray);
+        factoryWithNonNullZeroArray = new ArrayDataSerializableFactory(new ConstructorFunction[0]);
+        factoryWithNullArray = new ArrayDataSerializableFactory(null);
+    }
+
+    @Test
+    public void testCreate() throws Exception {
+        assertNotNull(factoryWithNonNullArray);
```

Comment:
Why are you asserting things which are created in the before? I would remove these 3 asserts.